cpp/build-support/upload-boost.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This assumes you've just run cpp/build-support/trim-boost.sh, so the file
+# to upload is at ${BOOST_FILE}/${BOOST_FILE}.tar.gz
+#
+# Also, you must have a bintray account on the "ursalabs" organization and
+# set the BINTRAY_USER and BINTRAY_APIKEY env vars.
```

Comment:
What do you think about hosting these tarballs as downloads from tagged GitHub releases (based on date or something)? Can do after this. Seems GitHub will be more reliable than Bintray

Reply:
I'm fine with whatever works; it was just quickest for me to use bintray because it's familiar. If someone wants to modify the script to put a tarball somewhere else, that's cool. 

They're not mutually exclusive, even: we can have multiple URLs as backups in the external project definition. So if you're worried about reliability, we can add redundancy.